### ISSUE TITLE
Blank JAVA_HOME with JDK < 8 in /usr/bin/java causes module-path to be enabled

### ISSUE THREAD

**Issue by rogerhu** [jruby/jruby-launcher#34]
The workaround right now is to define JAVA_HOME explicitly (https://github.com/jruby/jruby-launcher/commit/88e3c9613b63f1d4ac1db2a1730f2887422f7b17#r48855395) 

It very much depends on the [location of the java binary](https://github.com/jruby/jruby-launcher/blob/master/unixlauncher.cpp#L58-L62).  In this specific case, /usr/ somehow ends up being the directory, and because `/usr/lib/modules` exists, `--module-path` is triggered.

It looks like the same issue is here: https://github.com/headius/jruby/blob/ebf92b39391f04f06ebd6e3c54878f5a4bba4632/bin/jruby.bash#L139

If we have java in `/usr/bin/java`, it will resolve to `/usr/`:

```
 dirname $(dirname /usr/bin/java)
/usr
```

In other words, it will automatically search for `/usr/lib/modules` and find a path.  If you're using a JDK < 9, this will obviously introduce problems.

Seems like the solution is to specify JAVA_HOME always to avoid this issue.

**Comment by headius** [comment]
Clarifying here... setting JAVA_HOME to a real JDK or JRE location is always our recommended way to run, but since there are systems that do not set it *even when installing Java as a package* (as you have seen on CentOS) we must handle that case.

**Comment by headius** [comment]
jruby-launcher v1.1.15 has been released with the fixes from #35.

**Comment by headius** [comment]
@enebo I don't think any of my changes require an update of the Windows executables, but if you could double check I would appreciate it.
